[2024-06-16T00:55:09.057+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-16T00:55:09.114+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:52:32+00:00 [queued]>
[2024-06-16T00:55:09.130+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:52:32+00:00 [queued]>
[2024-06-16T00:55:09.131+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 6
[2024-06-16T00:55:09.156+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): avg_emission_by_year> on 2024-06-16 00:52:32+00:00
[2024-06-16T00:55:09.164+0000] {standard_task_runner.py:63} INFO - Started process 5372 to run task
[2024-06-16T00:55:09.170+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'avg_emission_by_year', 'manual__2024-06-16T00:52:32+00:00', '--job-id', '304', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmptr57ur5j']
[2024-06-16T00:55:09.174+0000] {standard_task_runner.py:91} INFO - Job 304: Subtask avg_emission_by_year
[2024-06-16T00:55:09.283+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-16T00:52:32+00:00 [running]> on host 6e26965f02f1
[2024-06-16T00:55:09.472+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='avg_emission_by_year' AIRFLOW_CTX_EXECUTION_DATE='2024-06-16T00:52:32+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-16T00:52:32+00:00'
[2024-06-16T00:55:09.474+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-16T00:55:09.475+0000] {summary_analysis.py:33} INFO - starting average pollutants emission calculation
[2024-06-16T00:55:09.478+0000] {summary_analysis.py:45} ERROR - error calculating average emission
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 36, in avg_emission_by_year
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 791, in read_json
    json_reader = JsonReader(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 904, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 944, in _get_data_from_filepath
    self.handles = get_handle(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 728, in get_handle
    ioargs = _get_filepath_or_buffer(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 472, in _get_filepath_or_buffer
    raise ValueError(msg)
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
[2024-06-16T00:55:09.480+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-16T00:55:09.480+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 36, in avg_emission_by_year
    val_data = pd.read_json(val_data)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 791, in read_json
    json_reader = JsonReader(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 904, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/json/_json.py", line 944, in _get_data_from_filepath
    self.handles = get_handle(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 728, in get_handle
    ioargs = _get_filepath_or_buffer(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 472, in _get_filepath_or_buffer
    raise ValueError(msg)
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
[2024-06-16T00:55:09.493+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=avg_emission_by_year, run_id=manual__2024-06-16T00:52:32+00:00, execution_date=20240616T005232, start_date=20240616T005509, end_date=20240616T005509
[2024-06-16T00:55:09.511+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 304 for task avg_emission_by_year (Invalid file path or buffer object type: <class 'NoneType'>; 5372)
[2024-06-16T00:55:09.543+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-16T00:55:09.579+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-16T00:55:09.582+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
